src/tests/ftest/util/slurm_utils.py

```diff
@@ -0,0 +1,194 @@
+    """
+    Generates a script for submitting a job to slurm.
+
+    path      --where to write the script file
+    name      --job name
+    output    --where to put the output (full path)
+    nodecount --number of compute nodes to execute on
+    cmds      --shell commands that are to be executed
+    sbatch    --dictionary containing other less often used parameters to
+                sbatch, e.g. mem:100
+    returns   --the full path of the script
+    """
+
+    if name is None or output is None or nodecount is None or cmds is None:
+        raise SlurmFailed("Bad parameters passed for slurm script.")
+
+    unique = random.randint(1, 100000)
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+    scriptfile = path + '/jobscript' + str(unique) + ".sh"
+
+    f = open(scriptfile, 'w')
+
+    # identify what be used to run this script
+    f.write("#!/bin/bash\n#\n")
+
+    # write the mandatory parameters
+    f.write("#SBATCH --job-name={}\n".format(name))
+    f.write("#SBATCH --output={}\n".format(output))
+    f.write("#SBATCH --nodes={}\n".format(nodecount))
+    f.write("#SBATCH --distribution=cyclic\n")
+
+    if sbatch:
+        for key, value in sbatch:
+            f.write("#SBATCH --{}={}\n".format(key, value))
+    f.write("\n")
+
+    # debug
+    f.write("echo \"nodes: \" $SLURM_JOB_NODELIST \"\"\n")
+    f.write("echo \"node count: \" $SLURM_JOB_NUM_NODES \" \"\n")
+    f.write("echo \"job name: \" $SLURM_JOB_NAME \"\"\n")
+
+    for cmd in cmds:
+        f.write("srun " + cmd)
+        f.write("\n")
+
+    f.close()
+    return scriptfile
+
+def run_slurm_cmd(cmd, name):
+
+    """
+        For very simple situations where you just want to run a simple
+        shell command via slurm this will do the trick.
+
+        cmd  --command that can be run from a shell
+        name --job name
+
+        returns --the job ID, which is used as a handle for other functions
+    """
+
+    daos_test_job = {"wrap": cmd, "job_name": name}
+    jobid = pyslurm.job().submit_batch_job(daos_test_job)
+    return jobid
+
+def run_slurm_script(script):
+
+    """
+        For running scripts, for example those created with write_slurm_script
+        function above.
+
+        script --a script file suitable to be run by slurm
+
+        returns --the job ID, which is used as a handle for other functions
+    """
+
+    cmd = "sbatch " + script
+    output = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
+    print output
+    if 'Submitted batch job' in output:
+        output_list = output.split()
+        if len(output_list) >= 4:
+            return output_list[3]
+
+    raise SlurmFailed("Batch job failed to start: {}".format(output))
+
+def get_job_output(handle, path):
+    """
+        Get the output from a slurm job.
+
+        handle   --slurm job id
+
+
+        returns  --string containing both stdout and stderr
+    """
+    pass
```

Comment:
`pass`?

Reply:
place holder for future function, deleted.